### ISSUE TITLE
[junit-platform] Comparison failures doesn't handle subclasses

### ISSUE THREAD

**Issue by kriegfrj** [bndtools/bnd#4338]
I've noticed while using AssertJ soft assertions that the GUI comparison feature doesn't seem to work when using the junit-platform tester.

Upon inspection I think I know why this is: the code in JUnitEclipseListener looks for exact matches on the class name and doesn't consider inheritance:
https://github.com/bndtools/bnd/blob/f03d29ab2740eec4308eef02675c54204506a441/biz.aQute.tester.junit-platform/src/aQute/tester/junit/platform/JUnitEclipseListener.java#L303-L305

AssertJ's soft assertions throw an `AssertJMultipleFailuresError`, which extends `org.opentest4j.MultipleFailuresError`.  But because none of the cases in the switch above exactly matches AssertJMultipleFailuresError, it doesn't get handled and so no actual/expected information is sent.

This should be a fairly simple fix.

**Comment by bjhargrave** [comment]
Can we fix quickly for 5.2?

**Comment by kriegfrj** [comment]
I can have a fix by Monday.

**Comment by kriegfrj** [comment]
Sorry, this might have to wait another 12 hours...

**Comment by bjhargrave** [comment]
> Sorry, this might have to wait another 12 hours...

OK, I have made 5.2.0.RC1 without this fix. Once the PR is merged into master, I can cherry-pick it back onto next for 5.2.0.RC2.

**Comment by kriegfrj** [comment]
Sorry for missing the deadline yesterday, the tests for this feature with the nested `MultipleFailuresError`s triggered a bug in Eclipse (since fixed: https://github.com/eclipse/eclipse.jdt.ui/commit/1d9fd32e50a8ea8d9fab54719135e05366cbc926#diff-0ae0a30fa8133368cc4ea5a6dafee799) which made debugging them difficult. Should have been much easier than that...
